script.js

```diff
@@ -1,4 +1,106 @@
+var playerResults = [];
+var clickCount = 0;
+var tempNum = [];
+var playerNumber = 1;
+var playerGroupNumber = [];
+
+// clickCount is to control the mode:
+// When clickCount equal to even number, it's "Dice-roll mode":
+// Generate 2 random numbers and show to players
+// When clickCount equal to odd number, it's "Push mode":
+// Finalise the number according to the order player choose, and push the number into array "playerResults"
+
+var getRandomIndex = function () {
+  // create a random number from zero to array length minus one.
+  // this number is in the range from the first
+  // index (zero) to the last index (array length minus one)
+  var randomIndex = Math.floor(Math.random() * 6) + 1;
+  return randomIndex;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var currentNumber;
+  var rollDice1;
+  var rollDice2;
+
+  // Dice-roll mode: clickCount equal to even number
+  if (clickCount % 2 == 0) {
+    rollDice1 = getRandomIndex();
+    rollDice2 = getRandomIndex();
+
+    // if those 2 random numbers equal to each other then just the "Push mode" and push the number into array directly.
+    if (rollDice1 == rollDice2) {
+      currentNumber = rollDice1 * 10 + rollDice2;
+      myOutputValue = `The number of the First Dice is ${rollDice1}; <br>The number of the Second Dice is ${rollDice2}. <br><br>The number for player ${playerNumber} is ${currentNumber}. <br><br>Next player please click "Submit" to roll the dice.`;
+      playerResults.push(currentNumber);
+      playerGroupNumber.push(playerNumber);
+      playerNumber++;
+      clickCount++;
+    } else {
+      myOutputValue = `The number of the First Dice is ${rollDice1}; <br>The number of the Second Dice is ${rollDice2}. <br><br>Press "1" for choosing "First dice number + Second dice number"; or <br>Press "2" for choosing "Second dice number + First dice number".`;
+
+      // the generated random numbers are temporarily stored in array "tempNum"
+      tempNum = [rollDice1, rollDice2];
+    }
+
+    // Push mode: clickCount equal to odd number
+  } else {
+    //When player choose "1":
+    if (input == 1) {
+      playerNember = clickCount;
+      currentNumber = tempNum[0] * 10 + tempNum[1];
```

Comment:
i think there's no need to have `tempNum`, you can just use the values of `rollDice1` and `rollDice2` to figure out what the `currentNumber` is right?

Reply:
Actually, I tried to use the values of  `rollDice1` and `rollDice2` directly but failed.
I think it is because those 2 random numbers are generated in the previous mode (when `clickCount` % 2 == 0).
Then they are not able to be accessed in the following "Push mode".
And that's why I set `tempNum` to carry the random numbers across the modes.